src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyInterceptionValidator.cs

```diff
@@ -8,15 +8,24 @@ namespace FakeItEasy.Creation.CastleDynamicProxy
 
     internal class CastleDynamicProxyInterceptionValidator : IMethodInterceptionValidator
     {
+        private const string NonVirtualMethodsCantBeIntercepted =
+            "Non-virtual or sealed members can not be intercepted. Only interface members and non-sealed virtual, overriding, and abstract members can be intercepted.";
+
         private readonly MethodInfoManager methodInfoManager;
 
         public CastleDynamicProxyInterceptionValidator(MethodInfoManager methodInfoManager)
         {
             this.methodInfoManager = methodInfoManager;
         }
 
-        public bool MethodCanBeInterceptedOnInstance(MethodInfo method, object? callTarget, [NotNullWhen(false)]out string? failReason)
+        public bool MethodCanBeInterceptedOnInstance(MethodInfo method, object? callTarget, [NotNullWhen(false)] out string? failReason)
         {
+            if (method.DeclaringType!.IsInterface && !method.IsVirtual)
+            {
+                failReason = NonVirtualMethodsCantBeIntercepted;
+                return false;
+            }

```

Comment:
I had to add this check here instead of in `GetReasonForWhyMethodCanNotBeIntercepted`, because `GetInvokedMethod` throws an exception (in `MethodInfoManager.GetMethodOnTypeThatImplementsInterfaceMethod`) because it can't find the sealed method in the interface map, and doesn't account for the case where it's not found. Fixing `GetMethodOnTypeThatImplementsInterfaceMethod` to return null in that case wouldn't help, because we wouldn't know why, and anyway `GetInvokedMethod` assumes it will receive a non-null result from `GetMethodOnTypeThatWillBeInvokedByMethodInfo`.